app/controllers/datastreams_controller.rb

```diff
@@ -48,7 +48,7 @@ def update
     end
 
     respond_to do |format|
-      format.any { redirect_to solr_document_path(params[:item_id]), notice: msg, flash: { error: truncate_flash(error_msg) } }
+      format.any { redirect_to solr_document_path(params[:item_id]), notice: msg, flash: { error: error_msg&.truncate(254) } }
     end
```

Comment:
rails has a built in method for truncation which we can use instead of writing our own method (but we need to guard against error_msg being nil in this case, which is what was causing the original bug)